src/usage/search.md

```diff
@@ -1,83 +1,136 @@
+For a given category, get the nearest 10 neighbors to a vector
+```sql
+SELECT 1 FROM items WHERE category_id = 1 ORDER BY embedding <#> '[0.5,0.5,0.5]' limit 10
+```
 
+## Query options
+
+Search options are specified by [PostgreSQL GUC](https://www.postgresql.org/docs/current/config-setting.html).
+
+Set `ivf` scan lists to 1 in session:
 ```sql
-SELECT * FROM items ORDER BY embedding <-> '[3,2,1]' LIMIT 5;
+SET vectors.ivf_nprobe=1;
+```
+
+Set `hnsw` search scope to 40 in transaction:
+```sql
+SET LOCAL vectors.hnsw_ef_search=40;
 ```
 
-It's recommended if your do **not** take advantages of
+Set search mode to `vbase` as system default:
+```sql 
+ALTER SYSTEM SET vectors.search_mode=vbase;
+```
 
-* database transaction
-* deletions without `VACUUM`
-* WHERE clauses and very complex SQL statements
+Query options for `ivf`:
 
-### `vbase`
+| Option                   | Type    | Range          | Default | Description                               |
+| ------------------------ | ------- | -------------- | ------- | ----------------------------------------- |
+| vectors.ivf_nprobe       | integer | `[1, 1000000]` | `10`    | Number of lists to scan.                  |
+
+Query options for `hnsw`:
+
+| Option                   | Type    | Range          | Default | Description                               |
+| ------------------------ | ------- | -------------- | ------- | ----------------------------------------- |
+| vectors.hnsw_ef_search   | integer | `[1, 65535]`   | `100`   | Search scope of HNSW.                     |
 
-`vbase` is another search mode. In this mode, vector indexes behave like a database index. In `vbase` mode, searching results become a stream and every time the database pulls a row, the vector index computes a row to return. It's quite different from an ordinary vector search if you are using a vector search library, such as *faiss*. The latter always wants to know how many results are needed before searching. The original idea comes from [VBASE: Unifying Online Vector Similarity Search and Relational Queries via Relaxed Monotonicity](https://www.usenix.org/conference/osdi23/presentation/zhang-qianxi).
+Query options for general:
 
-Assuming you are using HNSW algorithm, you may want the following SQL to work:
+| Option                   | Type    | Range              | Default   | Description                            |
+| ------------------------ | ------- | ------------------ | --------- | -------------------------------------- |
+| vectors.enable_index     | boolean |                    | `on`      | Enables or disables the query planner. |
+| vectors.search_mode      | enum    | `"basic", "vbase"` | `"basic"` | Search mode.                           |
+| vectors.enable_prefilter | boolean |                    | `on`      | Enables or disables the prefilter.     |
 
+# Advanced usage
```

Comment:
```suggestion
# Filter
```

Reply:
There is a section `filter`, this is advanced usage because in most cases user need not change `search mode` and `prefilter` at all.

Reply:
We should assume in most cases user need change `search mode` since filter is an advantage over `pgvector` for users.